test/test_data/snapd.py

```diff
@@ -0,0 +1,29 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+mapping_test_data = [
+    {"product": "snapd", "version": "2.56", "version_strings": ["snapd-2.56"]},
+]
+package_test_data = [
+    {
+        "url": "https://altlinux.pkgs.org/p10/classic-aarch64/snapd-2.56-alt1.aarch64.rpm.html",
```

Comment:
If I had to guess, your problem may be here in the url.  Typically the tool expects to slam together effectively  `url  + "/" + package_name` and have something that returns the download, but that doesn't work here because the url you specified isn't a download directory.   

https://altlinux.pkgs.org/p10/classic-aarch64/snapd-2.56-alt1.aarch64.rpm.html/snapd-2.56-alt1.aarch64.rpm gives an error page. You may be getting a weird and broken condensed download as a result (i.e. it's condensing the 404 page), which would break your tests but not what you're seeing locally when you try against a real package.

I'm not sure if we have other examples from `altlinux.pkgs.org`  but if you grep through test_data/*.py you might find some useful examples.

Reply:
noted, I'll get the url fixed but the main error encountered is the checker not able to find the the product in some other version package of this product (snapd-2.61.1-0.fc38.aarch64.rpm) where as when i am running the the tool with this particular checker installed it is able to detect the the product, I am so confused regarding that any thought?